linode/manager
# upcoming: [M3-7424] - Update top menu for parent, child, and proxy accounts

## Description 📝
With the addition of account switching capabilities, users will require a clear visual indicator in the Top Menu to easily discern the active account. 

**For parent users:** Parent user's username and company name are shown.
**For proxy users:** Parent's user's username and child company name are shown.
**For child users:** Child user's username and company name are shown.
**For regular users:** no changes from current design; we will continue to display only the username.

## Changes  🔄
- When a user has a `user_type` of `parent`, `proxy`, or `child`, the company name is displayed below the username. Which username and company name are displayed depends upon what type of user is logged in.
- Added unit test coverage for top menu.

## Preview 📷

| `user_type`  | Top Menu |
| ------- | ------- |
| parent/proxy/child | ![Screenshot 2023-12-27 at 1 27 24 PM](https://github.com/linode/manager/assets/114685994/c5a52ae0-b3f1-4f9f-b09b-33b549f041d8) |
| null (a regular user) | ![Screenshot 2023-12-27 at 1 27 51 PM](https://github.com/linode/manager/assets/114685994/cc24125c-0a0e-4cd4-ad23-7877623fe413) |

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Check out this PR and `yarn dev`.
- Make sure the Parent/Child Account Access feature flag is on and the mocks are on.

### Verification steps 
- I'm still trying to think of a way to test this fully without account switching functionality actually completed, because mocks need to be on but we'd also need to mock different responses to the same endpoint (/profile) to test a proxy account's parent username and child company name.
- We can verify:
   - When the current account's user type is `parent | proxy | child`, the company name displays below the username in the top menu.
   - When the current account's user type is `proxy`, a second request is made to the `/profile` to get the parent's profile data using a different auth token. (Easily visible in the RQ dev tools.)
   - If the current account's user is `parent | proxy | child` and does not have a company name, only the username is displayed in the top menu - no change from prod.
   -  When the current account's user type is `null` (change ` rest.get('*/account/users/:user',` in `serverHandlers.ts`), no company name is displayed in the top menu - no change from prod.
   - When the feature flag is OFF, no company name is displayed in the top menu - no change from prod.
- Unit tests pass:
```
yarn test UserMenu
```

## As an Author I have considered 🤔

*Check all that apply*
